pkgs/development/compilers/llvm/common/default.nix

```diff
@@ -74,7 +74,98 @@ let
       p:
       builtins.path {
         name = builtins.baseNameOf p;
-        path = "${metadata.versionDir}/${p}";
+        path =
```

Comment:
This big nested conditional is not great — I think maybe it indicates that getVersionFile is not the right abstraction.  Maybe it'd be better to just have a list of patches for each major LLVM version?

Reply:
Maybe but there's unconditional patches and some patches are the same between versions while having different siblings. I was thinking of doing an attr set and have it work like a table but I found it to be simpler to do it this way.

I was thinking of doing a PR to clean up the common default.nix so it wouldn't be as sloppy. Probably in that one, I could have an attr set of versions and values which could be a string or null which then matches into figuring out where that patch will be.

Reply:
> I was thinking of doing an attr set and have it work like a table

This is what I was thinking of.  Was the problem just that you'd have to repeat the patch of a patch multiple times to say that different versions have the same patch?  If so, I think that would be easier to understand overall.

Reply:
Maybe, I'm not sure.